Sources/SynchronizationManager.swift

```diff
@@ -135,8 +135,12 @@ public class SynchronizationManager: PersistenceIntegration {
         if let fetched = fetched?.first {
             persistable = fetched
         } else {
-            persistable = try! self.persistentStore.create(type: type)
-            persistable.id = asset.id
+            do {
+                persistable = try self.persistentStore.create(type: type)
+                persistable.id = asset.id
+            } catch let error {
+                fatalError("Could not create the Asset persistent store\n \(error)")
```

Comment:
I would leave `fatalError` here (so even crash in live world).  There is no guarantee that the program will work properly without the model. (which is expected to exist). The creation of the model should only fail in really rare cases. If it fails something else is really broken.
@loudmouth what do you think?